pycheribuild/projects/cross/cheribsd.py

```diff
@@ -1194,16 +1195,22 @@ def __init__(self, config: CheriConfig):
         self.extra_kernels_with_mfs = []
         if self.buildFpgaKernels:
             if self.compiling_for_mips(include_purecap=True):
-                if self.crosscompile_target.is_hybrid_or_purecap_cheri():
-                    prefix = "CHERI_DE4_"
+                if self.purecapKernel:
+                    if self.mfs_root_image:
+                        self.extra_kernels_with_mfs.append("CHERI_PURECAP_DE4_MFS_ROOT")
+                        self.extra_kernels_with_mfs.append("CHERI_PURECAP_DE4_MFS_ROOT_FUZZ")
+                        self.extra_kernels_with_mfs.append("CHERI_PURECAP_DE4_MFS_ROOT_BENCHMARK")
                 else:
-                    prefix = "BERI_DE4_"
+                    if self.crosscompile_target.is_hybrid_or_purecap_cheri():
+                        prefix = "CHERI_DE4_"
+                    else:
+                        prefix = "BERI_DE4_"
                     # TODO: build the benchmark kernels? TODO: NFSROOT?
-                for conf in ("USBROOT", "USBROOT_BENCHMARK", "NFSROOT"):
-                    self.extra_kernels.append(prefix + conf)
-                if self.mfs_root_image:
-                    self.extra_kernels_with_mfs.append(prefix + "MFS_ROOT")
-                    self.extra_kernels_with_mfs.append(prefix + "MFS_ROOT_BENCHMARK")
+                    for conf in ("USBROOT", "USBROOT_BENCHMARK", "NFSROOT"):
+                        self.extra_kernels.append(prefix + conf)
+                    if self.mfs_root_image:
+                        self.extra_kernels_with_mfs.append(prefix + "MFS_ROOT")
+                        self.extra_kernels_with_mfs.append(prefix + "MFS_ROOT_BENCHMARK")
```

Comment:
This logic can be shared with purecapKernel if you set the right prefix. Might as well also add a CHERI_DE4_MFS_ROOT_FUZZ rather than having to special-case CHERI_PURECAP_DE4_MFS_ROOT_FUZZ.